components/validator/challenge/fetcher.go

```diff
@@ -43,16 +49,14 @@ type ProofAndPair struct {
 	Pair  []*big.Int
 }
 
-func (f *Fetcher) FetchProofAndPair(blockNumber uint64) (*ProofAndPair, error) {
-	ctx, cancel := context.WithTimeout(context.Background(), f.timeout)
-	defer cancel()
-
-	blockNumberHex := fmt.Sprintf("0x%x", blockNumber)
-	f.logger.Info("received block number hex", "hex", blockNumberHex)
+func (f *Fetcher) FetchProofAndPair(ctx context.Context, trace string) (*ProofAndPair, error) {
+	cCtx, cCancel := context.WithTimeout(ctx, f.timeout)
+	defer cCancel()
 
-	response, err := f.Client.Prove(ctx, &pb.ProofRequest{BlockNumberHex: blockNumberHex})
+	// ProofType_AGG(4) is used to generate proof
+	response, err := f.Client.Prove(cCtx, trace, 4)
```

Comment:
Are there any use cases planned for the other proofTypes? 
It would be nice to see them specified as separate types.

Reply:
I heard that there is no plan for using other proof type, so I hardcoded this as 4.
I'll mark `NOTE` with reference.